<launch>
    <arg name="model_name" default="FastSAM-s.pt" />
    <arg name="image_topic" default="/camera/image_raw" />
    <arg name="device" default="cpu" />

    <node name="fastsam_ros" pkg="fastsam_ros" type="fastsam_node.py" output="screen">
        <remap from="~image_in" to="$(arg image_topic)" />        
        <remap from="~visualization" to="/fastsam/visualization" />
        <remap from="~segmentation_mask" to="/fastsam/mask" />
        
        <param name="model_name" value="$(arg model_name)" />
        <param name="device" value="$(arg device)" />
        <param name="imgsz" value="1024" />
        <param name="conf" value="0.4" />
        <param name="iou" value="0.9" />
        <param name="retina_masks" value="True" />
        
        <param name="prompt_mode" value="everything" />
    </node>
</launch>
